### ISSUE TITLE
FED srcIdExpected and srcIdReceived

### ISSUE THREAD

**Issue by ghost** [cmsdaq/DAQAggregator#24]
In the FEDs, while the srcIdExpected is always set to the value returned by the hardware FED's getSrcId() method at the initial object mapping, the srcIdReceived is probably not correctly set in the FED's updateFromFlashlist(...), as revealed by comparing Java DAQView to React DAQView (there are red notifications whenever the two srcIds differ in a given FED).

I fixed this by introducing the following conditional value setting, when updating FEDs from the FEROL_INPUT_STREAM flashlist:

`if (flashlistRow.get("WrongFEDIdDetected").asInt() == 0){
        this.srcIdReceived = this.srcIdExpected;
}else{
        this.srcIdReceived = flashlistRow.get("WrongFEDId").asInt();
}`

This works for most cases (although in some cases it does not...as Philipp noticed "The still affected FEDs seem to have some value masked"), but I would rather confirm that I correctly understand the semantics of WrongFEDIdDetected and WrongFEDId columns in the FEROL_INPUT_STREAM flashlist.

For info, there is also a flashlist field called "expectedFedId", but I am not currently using it, because the expected srcId is already filled from the hardware DB which, I assume, is correct. I also noticed that the HWDB-filled expected srcId is, on many cases, different from the value of this flashlist field, at a given FED. These cases are about ~220 out of ~640 FEDs, for which the updateFromFlashlist was called during a test. Total number of FEDs produced in the snapshot was 699.

Thank you in advance for your comments on this one.

**Comment by andreh12** [comment]
this is a bit surprising that there are so many mismatches. Could it be that these correspond to the FEROLs which have two FRLs connected ?

**Comment by Phil2812** [comment]
If the FED has its FRL masked (grey source id in the old daqview), there might not be an entry in the ferolInputStream flashlist for that FED (at least that was the case for the FEDs I just checked). In that case, the model is never populated with the data and the check you added is never executed. If you add an additional check (outside of the flashlist update) for the frlMasked flag and set the srcIdReceived to the srcIdExpected in case it is true, the remaining warnings should be gone from the new DAQView.

That the expected id in the flashlist is different from the one loaded from the HWCfgDB is strange. Did you check the flashlist dispatching for possible problems? Edit: I had a quick look and did not see any problem. We should meet to have a look at the test together.

**Comment by Phil2812** [comment]
While debugging we found out that there was indeed a problem with dispatching the flashlists, as the FEROL_INPUT_STREAM flashlist has separate rows per FED and the used matcher ignored the third column name (streamNumber) to match by. The other flashlists the matcher is used for only have one row per FRL, containing data for both FEDs (and it works fine for them). Michail is going to commit the matcher variation we created.
